src/osp/UserInputHandler.h

```diff
@@ -34,168 +34,164 @@
     uint8_t m_referenceCount;
 };
 
-struct ButtonVar
-{
-    enum class VarTrigger { HOLD = 0, PRESSED = 1 };
-    enum class VarOperator { OR = 0, AND = 1 };
-
-    ButtonVar(ButtonMap::iterator button, VarTrigger trigger,
-               bool invert, VarOperator nextOp);
-
-    ButtonVar(ButtonMap::iterator button, ButtonVarConfig cfg);
-
-    //        m_button(button), m_bits(bits) {}
-
-    // Bit 0: Trigger Mode
-    //   0. Always true when button is held
-    //   1. True only when button had just been pressed
-    // Bit 1: Invert
-    //   0. Detect hold down or pressing down
-    //   1. Invert, detect button up, or unpress instead
-    // Bit 2: Operator
-    //   0. OR with next condition
-    //   1. AND with next condition
-    // Bit 3: Varinate?
-    //   0. Last condition, varinate
-    //   1. Next condition exists
-    //uint8_t m_bits;
-
-    ButtonMap::iterator m_button;
-    VarTrigger m_trigger;
-    bool m_invert;
-    VarOperator m_nextOp;
-};
+using ButtonMap_t = std::map<int, ButtonRaw, std::less<>>;
 
-struct ButtonVarConfig
+/**
+ * @brief Mouse motion state
+ */
+struct MouseMotion
 {
-    using VarTrigger = ButtonVar::VarTrigger;
-    using VarOperator = ButtonVar::VarOperator;
+public:
+    Vector2i m_rawDelta;
+    Vector2 m_smoothDelta{ 0.0f };
 
-    constexpr ButtonVarConfig(int device, int devEnum, VarTrigger trigger,
-                               bool invert, VarOperator nextOp) :
-            m_device(device),
-            m_devEnum(devEnum),
-            m_trigger(trigger),
-            m_invert(invert),
-            m_nextOp(nextOp) {}
+    /*
+     * Mouse responsiveness - float between (0.0f, 1.0f]
+     * larger numbers -> less smooth, smaller numbers -> more floaty
+     * Recommend leaving this around 0.5
+    */
+    float m_responseFactor{ 0.5f };
 
-    int m_device;
-    int m_devEnum;
-    VarTrigger m_trigger;
-    bool m_invert;
-    VarOperator m_nextOp;
-    //uint8_t m_bits;
+    uint8_t m_referenceCount;
 };
 
-struct ButtonConfig
+/**
+ * @brief Raw scroll state
+ */
+struct ScrollRaw
 {
-    std::vector<ButtonVarConfig> m_press;
-    bool m_holdable;
-    bool m_enabled;
-    int m_index;
+    Vector2i offset;
+
+    uint8_t m_referenceCount;
 };
 
-// Primary owner of button state, one per button
-struct ButtonControl
-{
-    uint16_t m_referenceCount;
+//-----------------------------------------------------------------------------
 
-    // hold is true if just all the hold conditions are true.
-    // ignore the press/releases
+enum class EVarTrigger : uint8_t { HOLD = 0, PRESSED = 1 };
+enum class EVarOperator : uint8_t { OR = 0, AND = 1 };
 
-    bool m_holdable;
-    bool m_triggered, m_held;
-    //std::array<ButtonVar, 10> m_vars;
-    ButtonExpr m_exprPress;
-    ButtonExpr m_exprRelease;
+/**
+ * @brief An invidual term in a boolean expression representing a (raw button)
+ */
+struct ControlTerm
+{
+    ControlTerm(ButtonMap_t::iterator button, EVarTrigger trigger,
```

Comment:
For trivial constructors like this, you can just do direct member initialization.

E.g.

ControlTerm{button, trigger, invert, nextop};

Reply:
This lets you omit the constructor entirely.